vega_sim/environment/environment.py

```diff
@@ -446,4 +458,11 @@ def step(self, vega: VegaServiceNetwork) -> None:
             if self.random_agent_ordering
             else self.agents
         ):
-            agent.step(state)
+            try:
+                agent.step(state)
+            except:
+                msg = f"Agent '{agent.key_name}' failed to step."
+                if raise_on_error:
+                    raise RuntimeError(msg)
```

Comment:
We should catch and re-raise the error the `step` threw here or we'll hide it (so just use `except Exception as e` then raise that `e`)